### PR TITLE
Forced focus on captcha window

### PR DESCRIPTION
**Pull request by vasugamdha**
As mentioned in #133 
Added focus on the captcha window, and focus on the input box, too.🥂🥂

### CODE REVIEW

File: src/captcha.py [hunk 1]
```diff
@@ -18,13 +18,15 @@ def captcha_builder(resp):
 
     layout = [[sg.Image('captcha.gif')],
           [sg.Text("Enter Captcha Below")],
-          [sg.Input()],
+          [sg.Input(key='input')],
```
Comment (abhinavatai) [inline src/captcha.py:21]: Please set the Input field as text with key "input1": `sg.InputText('',key='input1')`

File: src/captcha.py [hunk 2]
```diff
@@ -18,13 +18,15 @@ def captcha_builder(resp):
...
           [sg.Button('Submit', bind_return_key=True)]]
 
-    window = sg.Window('Enter Captcha', layout)
+    window = sg.Window('Enter Captcha', layout, finalize=True)
```
Comment (abhinavatai) [inline src/captcha.py:24]: Also set use_default_focus=False: `sg.Window('Enter Captcha', layout, use_default_focus=False, finalize=True)`

File: src/captcha.py [hunk 3]
```diff
@@ -18,13 +18,15 @@ def captcha_builder(resp):
...
+    window.TKroot.focus_force()         # focus on window
+    window.Element('input').SetFocus()    # focus on field
```
Comment (abhinavatai) [inline src/captcha.py:26]: You get can get key like a dict and use set_focus instead: `window["input1"].set_focus()`

File: src/captcha.py [hunk 4]
```diff
@@ -18,13 +18,15 @@ def captcha_builder(resp):
...
     event, values = window.read()
     window.close()
-    return values[1]
+    return values['input']
```
Comment (abhinavatai) [inline src/captcha.py:29]: return `values["input1"]`

**Review by abhinavatai — COMMENTED** [review]
Please check comments.

**Comment by bombardier-gif** [comment]
@abhinavatai Could you resolve the comments and let me know if the code is good?
@vasugamdha Excellent contribution mate!

**Comment by vasugamdha** [comment]
> Please check comments.

@abhinavatai, thanks for the suggestions. But, the changes you are suggesting will work the same way as mine. In fact, I just booked with the code, changed as above, and it works completely fine. I might have missed your point of view, so please state the reason for your changes.  

> @vasugamdha Excellent contribution mate!

You're welcome!
🥂 to the community.

**Comment by vasugamdha** [comment]
@abhinavatai @bombardier-gif  You can check the changes yourself by running this [`captcha_test.py`](https://github.com/bombardier-gif/covid-vaccine-booking/blob/58338a5228d6581493172b838573894327bdeed3/tests/captcha_tests.py). It works well.

**Comment by abhinavatai** [comment]
> @abhinavatai @bombardier-gif You can check the changes yourself by running this [`captcha_tests.py`](https://github.com/bombardier-gif/covid-vaccine-booking/blob/58338a5228d6581493172b838573894327bdeed3/tests/captcha_tests.py). It works well.

Indeed. The focusing thing will work irrespective of the changes suggested. The changes I suggested are purely based on the purpose of code. I already implemented the focusing logic 2 days earlier on a request., that's why came with these changes.
Moreover, if your provide `use_default_focus=False` option, it will be more accurate since we are forcing the focus later on.

Also by using a test file you provided, it will not actually solve the purpose of testing.
The Reason is the captcha window opened in booking case is being opened by the booking script and it keeps require cursor focus on it, therefore the focus on captcha window is not set. 
And in your test case captcha window is being opened directly. It will always have focus irrespective of using force focus and set focus. You can removing those lines and test it, it will already have focus on input.

Thanks for your efforts.

**Comment by vasugamdha** [comment]
@abhinavatai I tried out your changes, but it is not working as I am trying to suggest.
There will be captchas popping one after the other, and at that time, you'll need to click there (for focusing on that window) every time. Instead, in my way, you have to enter the captchas continuously as it will autofocus to the text box.
Hey, I couldn't find your contact. Can you contact me on [t.me/vazug](t.me/vazug)?

**Comment by gitmnj** [comment]
> As mentioned in #133 
> 
> Added focus on the captcha window, and focus on the input box, too.🥂🥂

@vasugamdha Well done bro Nice Work done

Works for me even I booked slots for my 3-4 friends 

Very good contribution!!!

**Comment by vasugamdha** [comment]
Thanks, @gitmnj.🥂

**Comment by bombardier-gif** [comment]
Thanks a lot guys for working on this!

**Pull request merged**